go-controller/pkg/crd/userdefinednetwork/v1/cudn.go

```diff
@@ -100,6 +102,7 @@ type LocalnetConfig struct {
 	// The network will be assigned to pods who has the `k8s.v1.cni.cncf.io/networks` annotation in place pointing
 	// to subject.
 	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="layer2 topology config is immutable"
```

Comment:
I guess these messages need to be fixed to say that actual immutable field name instead of "layer2 topology"